zerver/lib/markdown/__init__.py

```diff
@@ -1595,10 +1595,55 @@ def __init__(self, compiled_re: Pattern, md: markdown.Markdown) -> None:
 
 
 class AutoLink(CompiledPattern):
+    def shorten_links(self, href: str) -> Optional[str]:
+        parts = urllib.parse.urlparse(href)
+        scheme, netloc, path, params, query, fragment = parts
+        text = None
+        if netloc in ["github.com", "www.github.com"]:
```

Comment:
Nobody uses www.github.com and GitHub itself doesn’t shorten www.github.com links, so we shouldn’t either. And we should check that the scheme is `https`.